BlockEntity/BECharcoalPit.cs

```diff
@@ -86,38 +90,33 @@ private void OnServerTick(float dt)
 
             if (state == 0) return;
 
-            BlockPos holePos = FindHoleInPit();
+            List<BlockPos> holes = FindHolesInPit();
 
-            if (holePos != null)
+            if (holes.Count > 0)
             {
+                Block fireblock = Api.World.GetBlock(new AssetLocation("fire"));
                 finishedAfterTotalHours = Api.World.Calendar.TotalHours + BurnHours;
 
-                BlockPos tmpPos = new BlockPos();
-                BlockFacing firefacing = null;
-
-                Block block = Api.World.BlockAccessor.GetBlock(holePos);
-                if (block.BlockId != 0 && block.BlockId != Block.BlockId)
+                foreach (BlockPos holePos in holes)
                 {
-                    foreach (BlockFacing facing in BlockFacing.ALLFACES)
+                    BlockPos firePos = holePos.Copy();
+
+                    Block block = Api.World.BlockAccessor.GetBlock(holePos);
+                    if (block.BlockId != 0 && block.BlockId != Block.BlockId)
                     {
-                        tmpPos.Set(holePos).Add(facing);
-                        if (Api.World.BlockAccessor.GetBlock(tmpPos).BlockId == 0)
+                        foreach (BlockFacing facing in BlockFacing.ALLFACES)
                         {
-                            holePos.Set(tmpPos);
-                            firefacing = facing;
-                            break;
+                            facing.IterateThruFacingOffsets(firePos);  // This must be the first command in the loop, to ensure all facings will be properly looped through regardless of any 'continue;' statements
+                            if (Api.World.BlockAccessor.GetBlock(firePos).BlockId == 0 && Api.World.Rand.NextDouble() > 0.9f)
+                            {
+                                Api.World.BlockAccessor.SetBlock(fireblock.BlockId, firePos);
+                                BlockEntity befire = Api.World.BlockAccessor.GetBlockEntity(firePos);
+                                befire?.GetBehavior<BEBehaviorBurning>()?.OnFirePlaced(facing, startedByPlayerUid);
+                            }
                         }
                     }
                 }
 
-                if (firefacing != null)
-                {
-                    Block fireblock = Api.World.GetBlock(new AssetLocation("fire"));
-                    Api.World.BlockAccessor.SetBlock(fireblock.BlockId, holePos);
-                    BlockEntity befire = Api.World.BlockAccessor.GetBlockEntity(holePos);
-                    befire?.GetBehavior<BEBehaviorBurning>()?.OnFirePlaced(firefacing, startedByPlayerUid);
-                }
-

```

Comment:
This now looks for every open hole and semi-randomly ignites them over time, keeping it from getting stuck without an obvious fire lighting up under specific circumstances.